src/api/scan/base_controller.go

```diff
@@ -110,118 +117,156 @@ func NewController() Controller {
-					}
-
-					// Recorded by error wrap and logged at the same time.
-					if err == nil {
-						err = e
-					} else {
-						err = errors.Wrap(e, err.Error())
-					}
-
-					logger.Error(errors.Wrap(e, "scan controller: scan"))
-					continue
-				}
+	for _, art := range artifacts {
+		trackID, producesMimes, err := bc.makeReportPlaceholder(ctx, r, art, options...)
+		if err != nil {
+			return err
+		}
 
-				producesMimes = append(producesMimes, pm)
-			}
+		if len(producesMimes) > 0 {
+			params = append(params, &Param{Artifact: art, TrackID: trackID, ProducesMimes: producesMimes})
+		}
+	}
 
-			break
+	var errs []error
+	for _, param := range params {
+		if err := bc.scanArtifact(ctx, r, param.Artifact, param.TrackID, param.ProducesMimes); err != nil {
+			log.Warningf("scan artifact %s@%s failed, error: %v", artifact.RepositoryName, artifact.Digest, err)
+			errs = append(errs, err)
 		}
 	}
 
-	// Scanner does not support scanning the given artifact.
-	if !matched {
-		return errors.Errorf("the configured scanner %s does not support scanning artifact with mime type %s", r.Name, artifact.MimeType)
+	// all scanning of the artifacts failed
+	if len(errs) == len(params) {
+		return fmt.Errorf("scan artifact %s@%s failed", artifact.RepositoryName, artifact.Digest)
+	}
+
+	return nil
+}
+
+func (bc *basicController) makeReportPlaceholder(ctx context.Context, r *scanner.Registration, art *ar.Artifact, options ...Option) (string, []string, error) {
+	trackID, err := bc.uuid()
+	if err != nil {
+		return "", nil, errors.Wrap(err, "scan controller: scan")
+	}
+
+	// Parse options
+	ops, err := parseOptions(options...)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "scan controller: scan")
 	}
 
-	// If all the record are created failed.
-	if len(producesMimes) == 0 {
-		// Return the last error
-		if statusConflict {
-			return errs.WithCode(errs.Conflict, errs.Wrap(err, "scan controller: scan"))
+	create := func(ctx context.Context, digest, registrationUUID, mimeType, trackID string, status job.Status) error {
+		reportPlaceholder := &scan.Report{
+			Digest:           digest,
+			RegistrationUUID: registrationUUID,
+			Status:           status.String(),
+			StatusCode:       status.Code(),
+			TrackID:          trackID,
+			MimeType:         mimeType,
+		}
+		// Set requester if it is specified
+		if len(ops.Requester) > 0 {
+			reportPlaceholder.Requester = ops.Requester
+		} else {
+			// Use the trackID as the requester
+			reportPlaceholder.Requester = trackID
 		}
 
-		return errors.Wrap(err, "scan controller: scan")
+		_, e := bc.manager.Create(reportPlaceholder)
+		return e
 	}
 
+	if HasCapability(r, art) {
+		var producesMimes []string
+
+		for _, pm := range r.GetProducesMimeTypes(art.ManifestMediaType) {
+			if err = create(ctx, art.Digest, r.UUID, pm, trackID, job.PendingStatus); err != nil {
+				return "", nil, err
+			}
+
+			producesMimes = append(producesMimes, pm)
+		}
+
+		if len(producesMimes) > 0 {
```

Comment:
why return when `len(producesMimes) > 0`?

Reply:
That means the report placeholders are all created success, so return directly. Otherwise the artifact can not be scanned by the scanner then just create a error report placeholder for the artifact.